compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -480,6 +481,17 @@ extern "C" LLVMTargetMachineRef LLVMRustCreateTargetMachine(
     // it prevents control flow from "falling through" into whatever code
     // happens to be laid out next in memory.
     Options.TrapUnreachable = true;
+    // But don't emit traps after other traps or no-returns unnecessarily.
+    std::string wasm = "wasm";
+    if (std::string(TripleStr).substr(0, wasm.length()) != wasm) {
+      Options.NoTrapAfterNoreturn = true;
+    } else {
+      // ...except for when targeting WebAssembly, because the NoTrapAfterNoreturn
+      // option causes bugs in the LLVM WebAssembly backend. You should be able to
+      // remove this when Rust's minimum supported LLVM version is >= 18
+      // https://github.com/llvm/llvm-project/pull/65876
+      Options.NoTrapAfterNoreturn = false;
```

Comment:
No need to explicitly set it to false. You can drop the else and move the comment above the if.